tophat/yvm
# refactor: switch install shell to javascript

## Description

Resolves #266
- Replaced the contents of `install.sh` with a call to `install.js`
- Now replaces lines in rc files to ensure they are to spec.

## DevQA

### DevQA Steps
1. Install script latest version
```sh
curl -fsSL https://raw.githubusercontent.com/tophat/yvm/install-js-script/scripts/install.js | node
```
- Launch new shell
- `yvm version` - should be latest version

2. Install script specified version
```sh
curl -fsSL https://raw.githubusercontent.com/tophat/yvm/install-js-script/scripts/install.js | INSTALL_VERSION=<version> node
```
- Launch new shell
- `yvm version` - should be specified version

3. Confirm previous install script path works
```sh
curl -fsSL https://raw.githubusercontent.com/tophat/yvm/install-js-script/scripts/install.sh | INSTALL_VERSION="v0.9.26" bash
```
- This will fail as this PR is not merged yet 😅

### Comments:

- Most of the bulk is from tests
